src/main/java/application/rest/StacksAccess.java

```diff
@@ -0,0 +1,762 @@
+		JSONArray activateStacksJA = convertMapToJSON(activateStacks);
+		JSONArray deletedStacksJA = convertMapToJSON(deletedStacks);
+
+		System.out.println("new curated stacks: "+newStacksJA);
+		System.out.println("activated stacks: "+activateStacksJA);
+		System.out.println("obsolete stacks: "+deletedStacks);
+
+
+
+		// log successful changes too!
+		try {
+			msg.put("new curated stacks", newStacksJA);
+			msg.put("activate stacks", activateStacksJA);
+			msg.put("obsolete stacks", deletedStacksJA);
+		} catch (Exception e) {
+			System.out.println("exception cause: " + e.getCause());
+			System.out.println("exception message: " + e.getMessage());
+			e.printStackTrace();
+		}
+		System.out.println("finishing refresh");
+		return Response.ok(msg).build();
+	}
+	
+	private String getDesiredState(List<Map> versionColls, List<Map> activateColls) {
+		String state=null;
+		if (activateColls!=null) {
+			for (Map map : versionColls) {
+				String name = (String) map.get("name");
+				name = name.trim();
+				boolean match = false;
+				for (Map map1 : activateColls) {
+					String name1 = (String) map1.get("name");
+					String desiredState = (String) map1.get("desiredState");
+					name1 = name1.trim();
+					if (name1.contentEquals(name)) {
+						state=desiredState;
+					}
+				}
+			}
+		}
+		return state;
+	}
+
+	private JSONArray convertMapToJSON(List<Map> list) {
+		JSONArray ja = new JSONArray();
+		for (Map m : list) {
+			JSONObject jo = new JSONObject();
+			jo.putAll(m);
+			ja.add(jo);
+		}
+		return ja;
+	}
+	
+
+	@DELETE
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("/stacks/{name}/versions/{version}")
+	public Response deActivateStack(@Context final HttpServletRequest request,
+			@PathParam("name") final String name, @PathParam("version") final String version) throws Exception {
+		// make call to kabanero to delete collection
+		ApiClient apiClient = KubeUtils.getApiClient();
+		StackApi api = new StackApi(apiClient);
+		String plural = "stacks";
+
+		JSONObject msg = new JSONObject();
+
+		try {
+			StackList fromKabanero = null;
+			try {
+				fromKabanero = api.listStacks(namespace, null, null, null);
+			} catch (ApiException e) {
+				e.printStackTrace();
+			}
+			// mapOneResource(ApiClient apiClient, String group, String version, String plural, String namespace, String name)
+			Stack kabStack = api.getStack(namespace, name);
+			System.out.println("*** reading stack object: "+kabStack);
+			if (kabStack==null) {
+				System.out.println("*** " + "Stack name: " + name + " 404 not found");
+				msg.put("status", "Stack name: " + name + " 404 not found");
+				return Response.status(400).entity(msg).build();
+			}
+			List<StackSpecVersions> kabSpecVersions=null;
+			if (version!=null) {
+				kabSpecVersions=StackUtils.getKabInstanceVersions(fromKabanero, name);
+
+				for (StackSpecVersions versionFromKab:kabSpecVersions) {
+					if (version.contentEquals(versionFromKab.getVersion())) {
```

Comment:
Do you need to print an error message if they supplied a version, but that version was not currently in the Stack CR spec?

Reply:
yes, good idea